google/cloud/storage/client.py

```diff
@@ -972,7 +973,14 @@ def generate_signed_post_policy_v4(
 
         # encode policy for signing
         policy = json.dumps(
-            {"conditions": conditions, "expiration": policy_expires.isoformat() + "Z"},
+            collections.OrderedDict(
+                sorted(
+                    {
+                        "conditions": conditions,
+                        "expiration": policy_expires.isoformat() + "Z",
+                    }.items()
+                )
+            ),
```

Comment:
`expiration` sometimes encoded before `conditions` - in such a cases tests are failing

Reply:
I'm a little concerned about imposing order in signing code given it's only used for conformance tests.

However, it doesn't change the outcome of expectation and will work as expected as a user.